packages/core/strapi/lib/Strapi.js

```diff
@@ -174,6 +175,9 @@ class Strapi {
         await this.load();
       }
 
+      // Swap type customField for underlying data type
+      convertCustomFieldType(this);
```

Comment:
I think you should be doing that between the register and the bootstrap lifecycles in the load not in the start :)